08_algorithms/job-runner.js

```diff
@@ -0,0 +1,124 @@
+class Node {
+    constructor(value, priority) {
+        this.value = value
+        this.priority = priority
+    }
+
+    log() {
+        console.log(`${this.value} with priority ${this.priority}`)
+    }
+}
+
+class PriorityQueue {
+    
+    constructor(){
+        this.values = []
+    }
+
+    swap(index1, index2) {
+        let temp = this.values[index1];
+        this.values[index1] = this.values[index2];
+        this.values[index2] = temp;
+    }
+
+    bubbleUp() {
+        let index = this.values.length - 1
+        while(index > 0){
+            let parentIndex = Math.floor((index - 1)/2);
+            if(this.values[parentIndex].priority > this.values[index].priority) {
+                this.swap(index, parentIndex);
+                index = parentIndex;
+            } else{
+                break;
+            }
+        }
+        return 0;
+    }
+
+    enqueue(value) {
+        this.values.push(value)
```

Comment:
for better performance, it’s preferable to put an element in its place, but not push into an array and then sort it. You can check it here https://www.geeksforgeeks.org/implementation-priority-queue-javascript/